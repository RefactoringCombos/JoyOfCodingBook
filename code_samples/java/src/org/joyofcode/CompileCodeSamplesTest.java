package org.joyofcode;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.macosx.DiffMergeReporter;
import org.junit.Test;

import com.spun.util.io.FileUtils;

@UseReporter(DiffMergeReporter.class)
public class CompileCodeSamplesTest
{
  @Test
  public void test() throws Throwable
  {
    String base = "/Users/llewellyn/Github/JoyOfCodingBook";
    String[] languages = {"java", "cpp"};
    CompileCodeSamples.now(base, languages, "joy_of_code.%s");
  }
  @Test
  public void testRegexForCodeBlock() throws Exception
  {
    String text = "# Introduce Lambda\n" + "\n" + "\n" + "     <codeblock Introduce.Lambda.Example.1 />\n"
        + "  <codeblock Introduce.Lambda.Example.2 />\n" + "  <codeblock Introduce.Lambda.Example.3 />\n" + "";
    Approvals.verify(CompileCodeSamples.swapCodeBlocks(text, t -> "found: " + t.group(1)));
  }
  @Test
  public void testRegexForCodetags() throws Exception
  {
    String text = FileUtils.readFile("src/org/joyofcode/samples/tests/TestLambda.java");
    Approvals.verify(CompileCodeSamples.findTagsInCodeFile(text));
  }
}
