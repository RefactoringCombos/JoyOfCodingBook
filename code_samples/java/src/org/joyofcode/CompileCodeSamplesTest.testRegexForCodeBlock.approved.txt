# Introduce Lambda
found: Introduce.Lambda.Example.1
found: Introduce.Lambda.Example.2
found: Introduce.Lambda.Example.3
